src/main/java/com/amazon/opendistroforelasticsearch/knn/index/KNNIndexCache.java

```diff
@@ -104,8 +105,10 @@ public synchronized void rebuild() {
      *
      * @param removalNotification key, value that got evicted.
      */
-    public void onRemoval(RemovalNotification<String, KNNIndex> removalNotification) {
-        try {            KNNIndex knnIndex = removalNotification.getValue();
+    private void onRemoval(RemovalNotification<String, KNNIndex> removalNotification) {
+            KNNIndex knnIndex = removalNotification.getValue();
```

Comment:
We should have exception handling here for other errors. If the segmentation faults occur, it does crash the process as we do not handle sigsegv signal

Reply:
As long as executor.execute() happens no matter what, I don't have much opinion (or knowledge) about the other parts. I do think it's critical that the submission of the gc() happens regardless of whether the other code throws, so if you think it's appropriate to wrap the rest of the code in try/catch (any exception), then I'd probably just suggest making sure to submit the gc() call first.

Reply:
My point is ```gc()``` can fail when we try to delete the memory not in use which would raise signal from the C library.  In this case whether gc() is called inside or outside try/catch it would have no effect.  

Agree on submitting the gc() call first